### DISCUSSION TITLE
NODE_ENV in .env not changing...

### DISCUSSION THREAD

**Discussion by valdimarein** [vercel/next.js#41239] [Help] [answered]
I have set NODE_ENV='development'

When I change NODE_ENV='production' 

Regardless of the value set in in .env. If I console.log() or display process.env.NODE_ENV in the browser, I always get 'development'.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

`NODE_ENV` is sort of a holy value, and Next.js will set it for you. 

What's more, I can tell you already that using "NODE_ENV=production" in development mode will crash, and that's about it. The reason being that there's tons of code that depend on these values. Such code loads development tools and others that enable for example, Hot Module Reloading.

Here's a very good read on how NODE_ENV is used by libraries, and it can help you understand why messing with it can lead to unpredictable behavior, https://overreacted.io/how-does-the-development-mode-work/.

Use a variable you can control yourself instead.

If you still want to see it crash, what you need to do is `NODE_ENV=production next dev` and that'll trick the process into using production mode for the development script.  https://nextjs.org/docs/messages/non-standard-node-env.

If you need to see production behaviour with unminified code, and non transpiled files, use source maps, https://nextjs.org/docs/advanced-features/source-maps.

**Comment by valdimarein** [comment]
That makes perfect sense. Thank you for your reply.

**Reply by icyJoseph** [reply]
Please don't forget to mark as answered, I think there was an old response from times where it kinda worked, like Next 10-ish, but that's not the case anymore.
